public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -416,7 +416,18 @@ function (angular, _, moment, dateMath, kbn, templatingVariable, CloudWatchAnnot
       return _.chain(targets)
       .map(function(target) {
         var dimensionKey = _.findKey(target.dimensions, function(v) {
-          return templateSrv.variableExists(v) && !_.has(scopedVars, templateSrv.getVariableName(v));
+          if (!templateSrv.variableExists(v)) {
+            return false;
+          }
+
+          var variableName;
+          while (variableName = templateSrv.getVariableName(v)) {
+            if (_.has(scopedVars, variableName)) {
+              return false;
+            }
+            v = v.replace('$' + variableName, '').replace('[[' + variableName + ']]', '');
```

Comment:
I need to get variable name list from dimValue.

If the specified dimValue includes multiple template variable like `$foo_$bar_baz`, this line remove the variable from dimValue.

1st iteration: remove `$foo` and value becomes `_$bar_$baz`.
2nd iterartion: remove `$bar` and value becomes `__$baz`.
3rd iteration: remove `$baz` and value becomes `___`, no template variable includes.

If templateSrv support like getVariableNames, I'll use that.